### ISSUE TITLE
[pair tag highlighter] Crash on ubuntu 14.04 32bits when typing <>

### ISSUE THREAD

**Issue by Laurent3170** [geany/geany-plugins#442]
When I type "<" it's automatically completed by ">" so I use the delete touch and it really often crashes. See[ #1056](https://github.com/geany/geany/issues/1056)

**Comment by b4n** [comment]
@Laurent3170 could you check #447 fixes it for you?  It strongly believe it does, but as I'm not able to reliably reproduce your scenario I'm not 100% sure.

**Comment by Laurent3170** [comment]
@b4n Ok for testing... But I don't know how to update the plugin via Geany. Can you post me a link where it's explained or should I do something else like "cloning and replace in my source" ?

**Comment by b4n** [comment]
Hum yeah you'd need to build the new version manually.  This requires some dependencies (*git*, *libgtk2.0-dev*, *make*, *automake*, *intltool* and *libtool*, the rest should be pulled in automatically I think), then:

```console
$ git clone https://github.com/b4n/geany-plugins/ -b pairtaghighlighter/crash-on-null-tag
$ ./autogen.sh --disable-all-plugins --enable-pairtaghighlighter
$ make -j2
```
and then either
```console
$ make install
```
to install in the system (will overwrite your current one probably, and might require super user privileges);
or a more manual approach to simply test without affecting your system, my using an alternative configuration directory and installing the plugin in it:

```console
$ mkdir -p /tmp/geany-temp-conf/plugins/
$ libtool --mode=install install pairtaghighlighter/src/.libs/pairtaghighlighter.so /tmp/geany-temp-conf/plugins/
$ geany -v -c /tmp/geany-temp-conf/
```

**Comment by Laurent3170** [comment]
It looks like a bit complicated for me... But I'll give a try in the evening and let you know. Thanks indeed

**Comment by b4n** [comment]
> So, I've cloned, `cd` to `geany-plugins` to find `autogen.sh`

Oops sorry I forgot that part indeed, but good you did it yourself.

> [… ran] `autogen.sh` but with `make -j2` I have […] "`No specified target and no makefile found. Break.`"

Hum, that means that *autogen.sh* failed for some reason, most likely a missing dependency.  Could you show the output of it?

**Comment by Laurent3170** [comment]
```
You should add the contents of '/usr/share/aclocal/intltool.m4' to 'aclocal.m4'.
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I build/cache -I build -I build/bundled -I geanypy/m4 --install
build/i18n.m4:20: warning: macro 'AM_GLIB_GNU_GETTEXT' not found in library
aclocal: installing 'build/cache/libtool.m4' from '/usr/share/aclocal/libtool.m4'
aclocal: installing 'build/cache/ltoptions.m4' from '/usr/share/aclocal/ltoptions.m4'
aclocal: installing 'build/cache/ltsugar.m4' from '/usr/share/aclocal/ltsugar.m4'
aclocal: installing 'build/cache/ltversion.m4' from '/usr/share/aclocal/ltversion.m4'
aclocal: installing 'build/cache/lt~obsolete.m4' from '/usr/share/aclocal/lt~obsolete.m4'
aclocal: installing 'build/cache/nls.m4' from '/usr/share/aclocal/nls.m4'
aclocal: installing 'build/cache/pkg.m4' from '/usr/share/aclocal/pkg.m4'
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in `.'.
libtoolize: copying file `./ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIR, `build/cache'.
libtoolize: copying file `build/cache/libtool.m4'
libtoolize: copying file `build/cache/ltoptions.m4'
libtoolize: copying file `build/cache/ltsugar.m4'
libtoolize: copying file `build/cache/ltversion.m4'
libtoolize: copying file `build/cache/lt~obsolete.m4'
build/i18n.m4:20: warning: macro 'AM_GLIB_GNU_GETTEXT' not found in library
autoreconf: running: /usr/bin/autoconf --force
configure:12908: error: possibly undefined macro: AM_GLIB_GNU_GETTEXT
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/bin/autoconf failed with exit status: 1
```

**Comment by b4n** [comment]
Have you installed the *libgtk2.0-dev* package?

**Comment by Laurent3170** [comment]
No it wasn't. Now it is (with a lot of dependencies ;-)) I've run the autogen which seems ok. If I run `make install` will it update my Geany ? If yes what do I risk ?

**Comment by b4n** [comment]
if you run `make install` you'll likely require `sudo` to do it, and it'll *replace* your current pairtaghighlighter plugin binary.  Meaning if you want to revert back to the Ubuntu one, you'll have to re-install the Ubuntu package (which will still be there, but with some files replaced).
Also, it'll install some additional files in `/usr/local`, which is not a big concern, and that you can remove with `make uninstall` (but that will also remove the replaced plugin binary, so you'd have to reinstall the Ubuntu package (yes, this time too) to get the Ubuntu binary back).

Basically, it's not a big problem but if you want to test only, you should probably rather install some place else :)

**Comment by Laurent3170** [comment]
Thanks. So I did make install. I restart Geany, and checks if it doesn't crash anymore ? How to be sure that new plugin is installed ?

**Comment by b4n** [comment]
if you did make install, you'll have the new plugin.  You can't really check (apart that the crash are gone), the best I can suggest is checking the date on the plugin executable:
```console
$ ls -l /usr/*/lib/geany/pairtaghighlighter.so
```

And yes, try and reproduce the crash you had, and tell me the result :)

**Comment by codebrainz** [comment]
> it'll replace your current pairtaghighlighter plugin binary

It shouldn't since Ubuntu packages go in `/usr` and Autoconf default prefix is `/usr/local` so it should just hide/override the one from the repos one until uninstalled.

>the best I can suggest is checking the date on the plugin executable

The .so filename used to be shown in the Plugin Manager dialog for each plugin. I think it's still shown in the tooltip if you hover the plugin items in the list, but since GTK+ tooltips are white-on-light-grey in KDE, I can't actually read any GTK+ tooltips here to be sure.

**Comment by Laurent3170** [comment]
I've worked a while doing the manipulation that causes crashes and nothing happened :smile: .

For the date, I did a `locate pairtaghighlighter.so` and found only one dated of 15th june
`ls -l /usr/lib/i386-linux-gnu/geany/pairtaghighlighter.so` --> `root root 44975 juin  15 20:36 /usr/lib/i386-linux-gnu/geany/pairtaghighlighter.so`.
Can you confirm that I can delete the folder `geany-plugins` created for the cloning. The answer should be yes... But as it's a bit of chinese for me ;-)

**Comment by b4n** [comment]
> It shouldn't since Ubuntu packages go in `/usr` and Autoconf default prefix is `/usr/local` so it should just hide/override the one from the repos one until uninstalled.

No, because we install the plugins where Geany will find them, even if it's outside `prefix`.  It's kind of questionable, but it's the way that makes things Just Work, and it can be overridden with `--with-geany-libdir`.

> I've worked a while doing the manipulation that causes crashes and nothing happened :smile: .

Good, thanks for testing!

> Can you confirm that I can delete the folder `geany-plugins` created for the cloning. The answer should be yes... But as it's a bit of chinese for me ;-)

Yes you can, but then you won't be able to uninstall the plugin automatically (the files installed in `/usr/local/`) i.e. when Ubuntu gets the fixed version.  It's not a big problem, and if you don't have too many things in there you should be able to delete it manually when needed.

**Comment by Laurent3170** [comment]
> Yes you can, but then you won't be able to uninstall the plugin automatically (the files installed in /usr/local/) i.e. when Ubuntu gets the fixed version. It's not a big problem, and if you don't have too many things in there you should be able to delete it manually when needed.

I have not well understood. It may be better (for my case) to uninstall the plugin and wait that it's populated in ubuntu ppa's. I use `http://ppa.launchpad.net/geany-dev/ppa/ubuntu`. If so, can you explain me how to uninstall and BYW congrats for solving the bug and thanks for all your time and explanations. I use Geany to build websites http://run.as and http://edlo.eu and it's very useful!

**Comment by b4n** [comment]
To uninstall properly, go back in the *geany-plugins* directory you cloned, then run `sudo make uninstall`. Then, to get back the Ubuntu version of pairtaghighliter (if you want it), do `sudo apt-get install --reinstall geany-plugin-pairtaghighlighter`.
Then you can safely remove the *geany-plugins* directory and the packages you installed to build it (at least *libtool* and *libgtk2.0-dev*, but you might know which ones you had to install and which ones you already had).

**Comment by Laurent3170** [comment]
I did and it worked perfectly! Now the plugin is dated 14th of march. I've kept lilbtool and co. I let you close the Issue if everything is ok for you as for me it is

**Comment by Laurent3170** [comment]
And I can confirm it was this causing the crash as I have reproduced and... It crashed ;-)

**Comment by b4n** [comment]
Fixed by merged #447

**Comment by Laurent3170** [comment]
Happy to see that it appears in 1.28. I've found a ppa: https://launchpad.net/~geany-dev/+archive/ubuntu/ppa but it's unofficial and for dev, do you know if an official stable ppa will come one day (I haven't found it) ?
